src/tooltip.js

```diff
@@ -24,17 +24,26 @@ export function defaultFlamegraphTooltip () {
             .attr('class', 'd3-flame-graph-tip')
     }
 
-    tip.show = function (d) {
+    tip.show = function (d, chartRightBoundary) {
         tooltip
             .style('display', 'block')
             .transition()
             .duration(200)
             .style('opacity', 1)
             .style('pointer-events', 'all')
 
+        const tooltipAreaWidth = tooltip.nodes()[0].getBoundingClientRect().width
+        const tooltipRightBoundary = event.pageX + 10 + tooltipAreaWidth
+
+        let tooltipLeft = event.pageX + 10
+
+        if (tooltipRightBoundary > chartRightBoundary) {
```

Comment:
Could we completely flip the tooltip to the right if boundary is exceeded? Not just shift so it fits?

Reply:
This is a good idea. Let me try...